### PR TITLE
More info on the StatusIndicator and add Cancel Search on Messages Tab

### PR DESCRIPTION
**Pull request by rjmasikome**
As per title, please let me know if anything lacking.

Cheers,
RJ

### CODE REVIEW

File: frontend/src/components/pages/topics/Tab.Messages.tsx [hunk 1]
```diff
@@ -293,6 +304,10 @@ export class TopicMessageView extends Component<{ topic: TopicDetail }> {
         }
     }
 
+    cancelSearch = () => {
+        api.stopMessageSearch();
```
Comment (rikimaru0345) [inline frontend/src/components/pages/topics/Tab.Messages.tsx:308]: you can use ()=>api.stopMessageSearch(); inline

File: frontend/src/state/backendApi.ts [hunk 2]
```diff
@@ -192,6 +192,9 @@ const apiStore = {
     Messages: [] as TopicMessage[],
     MessagesElapsedMs: null as null | number,
 
+    MessagesBytesConsumed: 0 as number,
```
Comment (rikimaru0345) [inline frontend/src/state/backendApi.ts:195]: no need for `as number` in this case

File: frontend/src/state/backendApi.ts [hunk 3]
```diff
@@ -234,10 +237,17 @@ const apiStore = {
...
+                case 'progressUpdate':
+                    this.MessagesBytesConsumed = msg.bytesConsumed;
+                    this.MessagesTotalConsumed = msg.messagesConsumed;
+                    break;
+
                 case 'done':
                     this.MessagesElapsedMs = msg.elapsedMs;
                     // this.MessageSearchCancelled = msg.isCancelled;
                     this.MessageSearchPhase = "Done";
+                    this.MessagesBytesConsumed = 0;
```
Comment (rikimaru0345) [inline frontend/src/state/backendApi.ts:249]: better to reset values to zero when a new message search starts (and/or when the websocket.closed event happens)

File: frontend/src/state/backendApi.ts [hunk 4]
```diff
@@ -253,8 +263,14 @@ const apiStore = {
...
                     let m = msg.message as TopicMessage;
 
-                    if (m.key && typeof m.key === 'string' && m.key.length > 0)
-                        m.key = atob(m.key); // unpack base64 encoded key
+                    if (m.key && typeof m.key === 'string' && m.key.length > 0) {
+                        try {
+                            m.key = atob(m.key); // unpack base64 encoded key
+                        } catch (error) {
+                            // Empty
+                            // Only unpack if the key is base64 based
```
Comment (rikimaru0345) [inline frontend/src/state/backendApi.ts:271]: did that happen for you? 
in any case, good that you added a catch handler 👍

Comment (rjmasikome) [reply frontend/src/state/backendApi.ts:270]: Yes, it happened to some certain topics in my case

**Pull request merged**